File: python/watermark/Watermark_remove.py
from aspose.pdf import (
    WatermarkArtifact,
    Artifact,
    Document,
    SaveFormat
)


def remove():
    path_source = "../../TestData/test_with_watermark.pdf"
    doc = Document(path_source)

    if doc.Pages[1].Artifacts[1].Subtype == Artifact.ArtifactSubtype.Watermark:
        doc.Pages[1].Artifacts.Delete(doc.Pages[1].Artifacts[1])

    #save result pdf to file
    doc.Save("remove_watermark.pdf", SaveFormat.PDF)
        

File: cpp/watermark/Watermark_remove.cpp
#include <iostream>

#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"
#include "Aspose.PDF.Cpp/Artifacts/ArtifactCollection.h"

#include "system/string.h"
#include "system/io/file.h"
#include "drawing/imaging/image_format.h"

using namespace System;
using namespace Aspose::Pdf;

void remove()
{
    String pathSource = u"../../TestData/test_with_watermark.pdf";
    auto doc = MakeObject<Document>(pathSource);

    if(doc->get_Pages()->idx_get(1)->get_Artifacts()->idx_get(1)->get_Subtype() == Aspose::Pdf::Artifact::ArtifactSubtype::Watermark)
    {
        doc->get_Pages()->idx_get(1)->get_Artifacts()->Delete(doc->get_Pages()->idx_get(1)->get_Artifacts()->idx_get(1));
    }

    //save result pdf to file
    doc->Save(u"test.pdf", SaveFormat::Pdf);
}
